service/src/main/java/bio/terra/landingzone/service/landingzone/azure/LandingZoneService.java

```diff
@@ -273,6 +276,41 @@ public List<UUID> listLandingZoneIds(BearerToken bearerToken, UUID billingProfil
         .toList();
   }
 
+  /**
+   * Lists landing zone record for a billing profile ID.
+   *
+   * @param bearerToken bearer token of the calling user.
+   * @param billingProfileId the billing profile ID to search.
+   * @return landing zone record.
+   */
+  public LandingZoneRecord getLandingZoneRecord(BearerToken bearerToken, UUID billingProfileId) {
+    // Call BPM to validate user access to billing profile.
+    bpmService.verifyUserAccess(bearerToken, billingProfileId);
+
+    var landingZoneRecord = getLandingZoneRecordByBillingProfile(bearerToken, billingProfileId);
+    return landingZoneRecord;
+  }
+
+  /**
+   * Lists landing zones for all billing profiles that the calling user has access to.
+   *
+   * @param bearerToken bearer token of the calling user.
+   * @return list of landing zone records.
+   */
+  public List<LandingZoneRecord> listLandingZoneRecords(BearerToken bearerToken) {
```

Comment:
As part of this, do we need the pre-existing `listLandingZoneIdsByTarget` and `listLandingZoneIds` methods anymore? Can we remove them in favor of the new methods?

Reply:
Yes, this is a plan - I'll remove those by Target as soon as are not used. I'll create an issue for that.

Reply:
TOAZ-256 created